demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1974 @@
+ * to wait for such a callback.
+ */
+#define MQTT_AGENT_MS_TO_WAIT_FOR_NOTIFICATION    ( 5000U )
+
+/**
+ * @brief The maximum back-off delay (in milliseconds) for retrying connection
+ * to server.
+ */
+#define CONNECTION_RETRY_MAX_BACKOFF_DELAY_MS     ( 5000U )
+
+/**
+ * @brief The base back-off delay (in milliseconds) to use for connection retry
+ * attempts.
+ */
+#define CONNECTION_RETRY_BACKOFF_BASE_MS          ( 500U )
+
+/**
+ * @brief The maximum number of retries for connecting to server.
+ */
+#define CONNECTION_RETRY_MAX_ATTEMPTS             ( 5U )
+
+/**
+ * @brief Number of milliseconds in a second.
+ */
+#define NUM_MILLISECONDS_IN_SECOND                ( 1000U )
+
+/**
+ * @brief Milliseconds per second.
+ */
+#define MILLISECONDS_PER_SECOND                   ( 1000U )
+
+/**
+ * @brief Milliseconds per FreeRTOS tick.
+ */
+#define MILLISECONDS_PER_TICK                     ( MILLISECONDS_PER_SECOND / configTICK_RATE_HZ )
+
+/**
+ * @brief Each compilation unit that consumes the NetworkContext must define it.
+ * It should contain a single pointer to the type of your desired transport.
+ * When using multiple transports in the same compilation unit, define this
+ * pointer as void *.
+ *
+ * @note Transport stacks are defined in amazon-freertos/libraries/abstractions/transport/secure_sockets/transport_secure_sockets.h.
+ */
+struct NetworkContext
+{
+    SecureSocketsTransportParams_t * pParams;
+};
+
+/**
+ * @brief Structure used to store the topic filter to ota callback mappings.
+ */
+typedef struct OtaTopicFilterCallback
+{
+    const char * pcTopicFilter;
+    uint16_t usTopicFilterLength;
+    IncomingPubCallback_t xCallback;
+} OtaTopicFilterCallback_t;
+
+/**
+ * @brief Defines the structure to use as the command callback context in this
+ * demo.
+ */
+struct MQTTAgentCommandContext
+{
+    MQTTStatus_t xReturnStatus;
+    TaskHandle_t xTaskToNotify;
+    uint32_t ulNotificationValue;
+    void * pArgs;
+};
+
+/**
+ * @brief The MQTT agent context.
+ * In case of sharing the mqtt connection with other demos using the MQTT agent,
+ * this context should be declared non-static so that's it shared across all
+ * demo files.
+ */
+static MQTTAgentContext_t xGlobalMqttAgentContext;
+
+/**
+ * @brief The buffer is used to hold the serialized packets for transmission to
+ * and from the transport interface.
+ */
+static uint8_t pucNetworkBuffer[ MQTT_AGENT_NETWORK_BUFFER_SIZE ];
+
+/**
+ * @brief FreeRTOS blocking queue to be used as MQTT Agent context.
+ */
+static MQTTAgentMessageContext_t xCommandQueue;
+
+/**
+ * @brief The interface context used to post commands to the agent.
+ * For FreeRTOS its implemented using a FreeRTOS blocking queue.
+ */
+static MQTTAgentMessageInterface_t xMessageInterface;
+
+/**
+ * @brief The global array of subscription elements.
+ *
+ * @note No thread safety is required to this array, since the updates the array
+ * elements are done only from one task at a time. The subscription manager
```

Comment:
Isn't it updated from the OTA demo task at lines 1822 and 1859, and the MQTT agent task at lines 1111 and 1141? That would make this comment incorrect.

Reply:
Thanks, updated comment.